mcs/class/System.ServiceModel/Test/System.ServiceModel.Dispatcher/Bug32886Test.cs

```diff
@@ -63,16 +63,19 @@ public class Bug32886
 				var client = new TempConvertSoapClient (binding, remoteAddress);
 
 				var wait = new ManualResetEvent (false);
+
 				client.CelsiusToFahrenheitCompleted += delegate (object o, CelsiusToFahrenheitCompletedEventArgs e) {
-					if (e.Error != null)
-						throw e.Error;
-					Assert.AreEqual ("76.1", e.Result, "#1");
-					wait.Set ();
+					try {
+						if (e.Error != null)
+							throw e.Error;
+						Assert.AreEqual ("76.1", e.Result, "#1");
```

Comment:
I think we should change our general approach, cause any exception thrown here will not be catched anywhere and just result in a "uncaught exceptions" in the NUnit output (e.g. try changing the assert condition here and you'll see it).

In other tests, I've seen the following pattern:

```csharp
Exception ex = null;
string result = null;
client.CelsiusToFahrenheitCompleted += delegate (object o, CelsiusToFahrenheitCompletedEventArgs e) {

    ex = e.Error;
    result = e.Result;
    wait.Set ();
}
wait.WaitOne ();
Assert.IsNull (ex);
Assert.AreEqual ("76.1", result, "#1");
```

This is much better IMHO.